stages/test/test_skopeo.py

```diff
@@ -33,3 +42,95 @@ def test_schema_validation_skopeo(stage_schema, test_data, expected_err):
     else:
         assert res.valid is False
         testutil.assert_jsonschema_error_contains(res, expected_err, expected_num_errs=1)
+
+
+@pytest.mark.skipif(os.getuid() != 0, reason="needs root")
+@pytest.mark.skipif(not has_executable("podman"), reason="no podman executable")
+@pytest.mark.parametrize("dest_type,local_output_path", [
+    ("dir", "/tmp/test-output-skopeo-dir"),
+    ("oci-archive", "/tmp/test-output-skopeo.tar"),
+    ("oci", "/tmp/test-output-skopeo-dir"),
+])
+def test_skopeo_dir(tmp_path, stage_module, dest_type, local_output_path):
+    with make_container(tmp_path, {"file1": "file1 from base"}) as base_tag:
+        with make_container(tmp_path, {"file1": "file1 from final layer"}, base_tag) as cont_tag:
+            # export for the container-deploy stage
+            fake_container_dst = tmp_path / "fake-container"
+            subprocess.check_call([
+                "podman", "save",
+                "--format=oci-archive",
+                f"--output={fake_container_dst}",
+                cont_tag,
+            ])
+
+    inputs = {
+        "images": {
+            # seems to be unused with fake_container_path?
+            "path": fake_container_dst,
+            "data": {
+                "archives": {
+                    fake_container_dst: {
+                        "format": "oci-archive",
+                        "name": cont_tag,
+                    },
+                },
+            },
+        },
+    }
+
+    output_dir = tmp_path / "output"
+    options = {
+        "destination": {
+            "type": dest_type,
+            "path": local_output_path
+        }
+    }
+
+    stage_module.main(inputs, output_dir, options)
+
+    assert output_dir.exists()
+
+    result = output_dir / local_output_path.strip("/")
+    assert_file_exists(result)
+
+    # Debugging in case of assertion failed
+    print("Output dir contains:")
+    for d in os.walk(output_dir):
+        print("\n".join([f"{d[0]}/{f}" for f in d[2]]))
+
+    if dest_type == "dir":
```

Comment:
I wonder if we can find a nicer way for this if/elif - it seems either those should be part of the parameters of the paramized test or separate tests. Given how different the checks are I would probably go with different tests and shared helpers. Like a "make_inputs" helper and a "make_container" helper etc.

Reply:
A `format -> expected_files` mapping in the test parameters would make this easier to read and clean up the if-chaining.

Reply:
Agree but I'd keep it as is, to get this merged soonish

Reply:
One strawman PR for the test weaks is now in https://github.com/osbuild/osbuild/pull/1761